src/scenes/level.js

```diff
@@ -148,30 +146,55 @@ export class LevelScene extends Phaser.Scene {
 
     enemyKilled(name) {
         // Update kill quest score
-        if (characterData.getEnemiesKilled(this.currentLevel, name) < this.killQuest) {
+        if (
+            characterData.getEnemiesKilled(this.currentLevel, name) <
+            this.questAmounts[name][this.questAmounts[name].length - 1]
+        ) {
```

Comment:
I think pulling out these conditions into two variables would help the readability here. Took me a bit to see what you were doing.

Maybe:

```js
const enemiesKilled = characterData.getEnemiesKilled(this.currentLevel, name);
const maxKillCount = this.questAmounts[name][ this.questAmounts[name].length - 1 ];
if (enemiesKilled < maxKillCount) {

```

Reply:
[Updated](https://github.com/EthanVieira/rs-clicker/pull/47/commits/0d1df380ee541e72d588204b3ac45a7aa2cea81e)!